File: server/Models/Comment.cs
using System.ComponentModel.DataAnnotations;

namespace allspice.Models;

public class Comment : RepoItem<int>
{
  [MaxLength(5000)] public string Body { get; set; }
  [MaxLength(255)] public string CreatorId { get; set; }
  public Profile Creator { get; set; }
  public List<string> LikeIds { get; set; }
}

File: client/src/models/Comment.js
import { Profile } from "./Account.js";
import { DatabaseItem } from "./DatabaseItem.js";


export class Comment extends DatabaseItem{
  /**
   * @typedef CommentData
   * @property {number} id
   * @property {Date} createdAt
   * @property {Date} updatedAt
   * @property {string} body
   * @property {string} creatorId
   * @property {Profile} creator
   * @param {CommentData} data 
   */
  constructor(data){
    super(data)
    this.body = data.body
    this.creatorId = data.creatorId
    this.creator = data.creator
  }
}